packages/shared/src/hooks/feed/useFeedName.ts

```diff
@@ -0,0 +1,34 @@
+import { MainFeedPage } from '../../components/utilities';
+import { useFeature } from '../../components/GrowthBookProvider';
+import { Features } from '../../lib/featureManagement';
+import { SearchExperiment } from '../../lib/featureValues';
+
+interface UseFeedNameProps {
+  isSearchOn: boolean;
+  feedName: MainFeedPage;
+}
+
+interface UseFeedName {
+  isUpvoted: boolean;
+  isSortableFeed: boolean;
+}
+
+const sortableFeeds = [MainFeedPage.Popular, MainFeedPage.MyFeed];
+
+export const useFeedName = ({
+  feedName,
+  isSearchOn,
+}: UseFeedNameProps): UseFeedName => {
+  const searchVersion = useFeature(Features.Search);
+
+  return {
+    isUpvoted:
+      searchVersion === SearchExperiment.Control
+        ? !isSearchOn && feedName === 'upvoted'
```

Comment:
Can't we rather abstract the logic above?
To say isSearchOn is false//null when it's not control?

Reply:
That `isSearchOn` is coming from way up the tree based on a state. Can you provide a simple example of how you envision it to be?